### DISCUSSION TITLE
Changing/switching the name of the `$$is_a_module` property

### DISCUSSION THREAD

**Discussion by elia** [opal/opal#2210] [General]
_originated in: https://github.com/opal/opal/pull/1653#discussion_r110545242_

---

- `$$is_a_module` is true for all modules (+subclasses), i.e. whenever `is_a? Module` is `true`
- `$$is_module` is true for modules and false for classes
- `$$is_class` is true for classes and not for modules

This can lead to some confusion so it's better to keep just two of them or rename one of the three

cc @meh @opal/core

**Comment by iliabylich** [comment]
I like @meh's idea to have `$$is_namespace`. Having `is_a_module` and `is_module` is confusing.

**Comment by iliabylich** [comment]
By the way, can we make it a runtime helper? Like `Opal.truthy?`.
We could compile`Opal.namespace?(obj)` to something like `obj.$$is_class || obj.$$is_module` because it seems that it's not necessary to have all 3 props. Of course, it requires us to write this JS manually in `runtime.js` (then maybe it's even better to make a JS method `Opal.is_namespace(obj)`)

**Comment by meh** [comment]
Yeah we could actually go with a bunch of helpers, since they're really short they're 100% going to get inlined either way.

@elia raised some good points against `namespace`, there seem to be a bunch of cases where `is_module` and `is_class` are checked with logic other than AND.

**Comment by iliabylich** [comment]
@elia @meh From what I see everywhere is:
1. `$$is_module`
2. `$$is_class`
3. `$$is_module || $$is_class` -> `$$is_namespace`
4. `!$$is_module && !$$is_class` -> `!($$is_module || $$is_class)` -> `!$$is_namespace` (inverted case #3)

Looks like we can go with one runtime helper.
